### PR TITLE
Add Attribute conflation type

### PR DESCRIPTION
**Pull request by putipong**
ref #1276

### CODE REVIEW

**Comment by brianhatchl** [comment]
The code changes look fine, but it seems like the advanced options for Attribute conflation are being used when I select Differential or Differential w/Tags.  I think we want to be using the defaults.

Here's a comparison of what Hoot 1 vs 2x send for a Differential conflation job:

### Hoot 1
```
"-D "map.cleaner.transforms=hoot::ReprojectToPlanarOp;hoot::DuplicateWayRemover;hoot::SuperfluousWayRemover;hoot::IntersectionSplitter;hoot::UnlikelyIntersectionRemover;hoot::DualWaySplitter;hoot::ImpliedDividedMarker;hoot::DuplicateNameRemover;hoot::SmallWayMerger;hoot::RemoveEmptyAreasVisitor;hoot::RemoveDuplicateRelationMembersVisitor;hoot::RelationCircularRefRemover;hoot::RemoveEmptyRelationsOp;hoot::RemoveDuplicateAreaVisitor;hoot::NoInformationElementRemover;hoot::CornerSplitter;hoot::RubberSheet" -D "small.way.merger.threshold=15" -D "rubber.sheet.ref=true" -D "rubber.sheet.fail.when.minimum.tie.points.not.found=false" -D "rubber.sheet.log.missing.requirements.as.warning=true" -D "rubber.sheet.minimum.ties=4" -D "add.review.tags.to.features=false" -D "autocorrect.options=true" -D "element.cache.size.node=10000000" -D "element.cache.size.relation=2000000" -D "element.cache.size.way=2000000" -D "duplicate.name.case.sensitive=true" -D "edge.distance.extractor.spacing=5.0" -D "token.keep.non.words=false" -D "levenshtein.distance.alpha=1.15" -D "merge.nearby.nodes.distance=1.0" -D "token.min.size=3" -D "node.matcher.strictness=2" -D "ogr.esri.fcsubtype=true" -D "ogr.split.o2s=false" -D "ogr.thematic.structure=true" -D "translate.string.distance.tokenize=true" -D "weighted.metric.distance.extractor.search.radius=-1.0" -D "unify.optimizer.time.limit=60" -D "search.radius.highway=-1" -D "way.merger.min.split.size=5" -D "match.creators=hoot::NetworkMatchCreator;hoot::ScriptMatchCreator,Area.js;hoot::BuildingMatchCreator;hoot::ScriptMatchCreator,PoiGeneric.js;hoot::PoiPolygonMatchCreator;hoot::ScriptMatchCreator,PowerLine.js;hoot::ScriptMatchCreator,Railway.js;hoot::ScriptMatchCreator,LinearWaterway.js" -D "merger.creators=hoot::NetworkMergerCreator;hoot::ScriptMergerCreator;hoot::BuildingMergerCreator;hoot::PoiPolygonMergerCreator" -D "network.matcher=hoot::ConflictsNetworkMatcher" -D "building.review.if.secondary.newer=false" -D "building.date.format=yyyy-MM-ddTHH:mm" -D "building.date.tag.key=source:date" -D "building.keep.more.complex.geometry.when.auto.merging=true" -D "building.review.matches.other.than.one.to.one=false" -D "poi.ignore.type.if.name.present=false" -D "poi.polygon.address.match.enabled=false" -D "poi.polygon.auto.merge.many.poi.to.one.poly.matches=false" -D "poi.polygon.disable.same.source.conflation=false" -D "poi.polygon.disable.same.source.conflation.match.tag.key.prefix.only=true" -D "poi.polygon.keep.closest.matches.only=false" -D "poi.polygon.match.distance.threshold=5.0" -D "poi.polygon.match.evidence.threshold=3" -D "poi.polygon.name.score.threshold=0.8" -D "poi.polygon.phone.number.match.enabled=true" -D "phone.number.region.code=US" -D "phone.number.search.in.text=false" -D "poi.polygon.promote.points.with.addresses.to.pois=false" -D "poi.polygon.review.distance.threshold=125.0" -D "poi.polygon.review.evidence.threshold=1" -D "poi.polygon.review.multiuse.buildings=false" -D "poi.polygon.source.tag.key=source" -D "poi.polygon.type.score.threshold=0.7" -D "power.line.auto.calc.search.radius=true" -D "power.line.matcher.distance.weight.coefficient=0.01" -D "power.line.matcher.max.angle=155.0" -D "power.line.subline.matcher=hoot::FrechetSublineMatcher" -D "railway.subline.matcher=hoot::MaximalSublineMatcher" -D "railway.matcher.max.angle=90.0" -D "waterway.angle.sample.distance=20.0" -D "waterway.auto.calc.search.radius=true" -D "waterway.rubber.sheet.ref=true" -D "waterway.rubber.sheet.minimum.ties=5" -D "waterway.matcher.heading.delta=150.0" -D "waterway.matcher.max.angle=90.0" -D "search.radius.waterway=-1" -D "waterway.subline.matcher=hoot::MaximalSublineMatcher""
```

### Hoot 2x
```
"-D "map.cleaner.transforms=hoot::ReprojectToPlanarOp;hoot::DuplicateWayRemover;hoot::SuperfluousWayRemover;hoot::IntersectionSplitter;hoot::UnlikelyIntersectionRemover;hoot::DualWaySplitter;hoot::ImpliedDividedMarker;hoot::DuplicateNameRemover;hoot::SmallWayMerger;hoot::RemoveEmptyAreasVisitor;hoot::RemoveDuplicateRelationMembersVisitor;hoot::RelationCircularRefRemover;hoot::RemoveEmptyRelationsOp;hoot::RemoveDuplicateAreaVisitor;hoot::NoInformationElementRemover;hoot::CornerSplitter" -D "small.way.merger.threshold=15" -D "rubber.sheet.ref=true" -D "rubber.sheet.log.missing.requirements.as.warning=true" -D "rubber.sheet.minimum.ties=4" -D "autocorrect.options=true" -D "element.cache.size.node=10000000" -D "element.cache.size.relation=2000000" -D "element.cache.size.way=2000000" -D "duplicate.name.case.sensitive=true" -D "edge.distance.extractor.spacing=5.0" -D "levenshtein.distance.alpha=1.15" -D "merge.nearby.nodes.distance=1.0" -D "token.min.size=3" -D "node.matcher.strictness=2" -D "ogr.esri.fcsubtype=true" -D "ogr.thematic.structure=true" -D "translate.string.distance.tokenize=true" -D "weighted.metric.distance.extractor.search.radius=-1.0" -D "unify.optimizer.time.limit=60" -D "conflate.post.ops=hoot::RemoveUnknown2Visitor" -D "search.radius.highway=-1" -D "way.merger.min.split.size=5" -D "match.creators=hoot::NetworkMatchCreator;hoot::ScriptMatchCreator,Area.js;hoot::BuildingMatchCreator;hoot::ScriptMatchCreator,PoiGeneric.js;hoot::PoiPolygonMatchCreator;hoot::ScriptMatchCreator,PowerLine.js;hoot::ScriptMatchCreator,Railway.js;hoot::ScriptMatchCreator,LinearWaterway.js" -D "merger.creators=hoot::NetworkMergerCreator;hoot::ScriptMergerCreator;hoot::BuildingMergerCreator;hoot::PoiPolygonMergerCreator" -D "network.matcher=hoot::ConflictsNetworkMatcher" -D "building.date.format=yyyy-MM-ddTHH:mm" -D "building.date.tag.key=source:date" -D "building.keep.more.complex.geometry.when.auto.merging=true" -D "address.additional.tag.keys=" -D "poi.polygon.disable.same.source.conflation.match.tag.key.prefix.only=true" -D "poi.polygon.match.distance.threshold=5.0" -D "poi.polygon.match.evidence.threshold=3" -D "poi.polygon.name.score.threshold=0.8" -D "phone.number.additional.tag.keys=" -D "poi.polygon.phone.number.match.enabled=true" -D "phone.number.region.code=US" -D "poi.polygon.review.distance.threshold=125.0" -D "poi.polygon.review.evidence.threshold=1" -D "poi.polygon.review.if.matched.types=" -D "poi.polygon.source.tag.key=source" -D "poi.polygon.type.score.threshold=0.7" -D "power.line.auto.calc.search.radius=true" -D "power.line.matcher.distance.weight.coefficient=0.01" -D "power.line.matcher.max.angle=155.0" -D "power.line.subline.matcher=hoot::FrechetSublineMatcher" -D "railway.subline.matcher=hoot::MaximalSublineMatcher" -D "railway.matcher.max.angle=90.0" -D "waterway.angle.sample.distance=20.0" -D "waterway.auto.calc.search.radius=true" -D "waterway.rubber.sheet.ref=true" -D "waterway.rubber.sheet.minimum.ties=5" -D "waterway.matcher.heading.delta=150.0" -D "waterway.matcher.max.angle=90.0" -D "search.radius.waterway=-1" -D "waterway.subline.matcher=hoot::MaximalSublineMatcher""
```

File: modules/Hoot/ui/sidebar/advancedOpts/advancedOpts.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ import { d3combobox }   from '../../../../lib/hoot/d3.combobox';
 export default class AdvancedOpts {
     constructor() {
         this.sidebar         = d3.select( '#hoot-sidebar' );
-        this.optTypes        = [ 'custom', 'horizontal', 'reference', 'diffConflator', 'diffTags'];
+        this.optTypes        = [ 'custom', 'horizontal', 'average', 'reference', 'diff', 'diffTags', 'attribute' ];
```
Comment (brianhatchl) [inline modules/Hoot/ui/sidebar/advancedOpts/advancedOpts.js]: we intentionally removed `average` in another PR

File: modules/Hoot/ui/sidebar/advancedOpts/advancedOpts.js [hunk 2]
```diff
@@ -24,16 +24,16 @@ export default class AdvancedOpts {
...
-        // console.log( allOpts );
+        let allOpts = await Promise.all( _map( this.optTypes, type => Hoot.api.getAdvancedOptions( type ) ) );
 
         this.advancedOptions = {
             base: allOpts[ 0 ],
             horizontal: allOpts[ 1 ],
-            reference: allOpts[ 2 ],
-            diffConflator: allOpts[ 3 ],
-            diffTags: allOpts[ 4 ]
+            average: allOpts[ 2 ],
```
Comment (brianhatchl) [inline modules/Hoot/ui/sidebar/advancedOpts/advancedOpts.js]: we intentionally removed average in another PR

**Comment by brianhatchl** [comment]
Behavior looks good to me for Attribute, Differential, and Differential w/ Tags.
Remove the Average conflation references from the bad merge conflict resolution and this will be :+1:

**Comment by putipong** [comment]
average conflation has been removed

**Pull request merged**